### DISCUSSION TITLE
How can I achieve this Nested Dynamic Route example?

### DISCUSSION THREAD

**Discussion by amroczeK** [vercel/next.js#40462] [Help]
I have developed a website in NextJs that sells Wheelchair products, there are different categories of wheelchairs and the categories can have subcategories.

E.g. One category of wheelchair is Power Wheelchair, they can be Rear Wheel Drive, Front Wheel Drive or Mid Wheel Drive, being the subcategories of Power Wheelchair.

I am using getStaticPaths and getStaticProps in my existing file structure below before adding subcategories.

```
pages/
├─ products/
| └─ [category]
| | [slug].tsx
| | index.tsx
| index.tsx
```

The above produces e.g.:
/products/power-wheelchairs <--- page listing all products of this category
/products/power-wheelchairs/permobil-m300-corpus-hd <--- specific product of that category

### This is the routing I am trying to achieve with subcategories: 
Note: Everything in square bracket is dynamic route. I am using getStaticPaths and getStaticProps.

/products/[category] <--- lists all products for that category
/products/[category]/[product] <--- page for specific product in that category
/products/[category]/[subcategory] <--- lists all products for subcategory in that category
/products/[category]/[subcategory]/[product] <--- page for specific product in subcategory of that category

I have tried the following folder structure:
```
pages/
├─ products/
| └─ [category]
| |   └─ [subcategory]
| |    | [[...slug]].tsx
| |    | index.tsx
| | index.tsx
| index.tsx
```

But what I found was that /products/[category]/[product] would get rendered inside of the index.tsx in [subcategory] and not the [[..slug]].tsx. I have also tried other which ways but can't get the desired structure, does anyone have any recommendations?

I am considering removing the subcategory folder and combining it's index.tsx and the [slug].tsx into one catch all route [...slug], then conditionally render the page depending on the type of page... but I won't know if the slug refers to a subcategory or product for these two route examples:

/products/[category]/[product]
/products/[category]/[subcategory]

```
pages/
├─ products/
| └─ [category]
| | [...slug].tsx
| | index.tsx
| index.tsx
```

**Comment by akash-gajjar** [comment]
@amroczeK maybe you can use subcategory with default value of `all`
```diff
  /products/[category] <--- lists all products for that category
- /products/[category]/[product] <--- page for specific product in that category
+ /products/[category]/all/[product] <--- page for specific product in that category
  /products/[category]/[subcategory] <--- lists all products for subcategory in that category
  /products/[category]/[subcategory]/[product] <--- page for specific product in subcategory of that category
```
